src/Artifacts.jl

```diff
@@ -0,0 +1,589 @@
+module Artifacts
+
+import Base: get, SHA1
+import ..depots1
+import ..GitTools: tree_hash, set_readonly
+using ..BinaryPlatforms
+import ..TOML
+import ..Types: parse_toml
+import ..PlatformEngines: download_verify_unpack, probe_platform_engines!, package
+
+export create_artifact, artifact_exists, artifact_path, remove_artifact, verify_artifact,
+       artifact_meta, artifact_hash, bind_artifact, unbind_artifact, download_artifact,
+       find_artifact_toml, ensure_artifact_installed, @artifact_str, archive_artifact
+
+## Philosophy of Artifacts:
+#
+# - At the lowest level, artifacts are content-addressed chunks of data.  Similarly to
+#   how packages are identified by UUIDs but have human-readable mappings through the
+#   registry, artifacts are identified by their SHA1 git tree hashes, but have human-
+#   readable mappings through `Artifact.toml` files.
```

Comment:
For packages we also allow `JuliaProject.toml`. For consistency, it might be worth thinking about allowing `JuliaArtifact.toml` but not sure if it is worth the churn.

Reply:
I'm going to rename to `Artifacts.toml` anyway, so why not.

Reply:
Done